### PR TITLE
More release automation

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -240,3 +242,28 @@ jobs:
...
+
+  create-pull-request-against-main:
+    needs: release
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.3.4
+
+      - name: Bump download link version
+        run: |
+          sed -Ei "s,https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v[0-9]+.[0-9]+.[0-9]+/,https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v${{ needs.release.outputs.release-version }}/," README.md
```
Comment (anuraaga) [inline .github/workflows/release.yml:254]: Was wondering whether for main branch it doesn't make sense to just use `latest`

Comment (trask) [reply .github/workflows/release.yml:254]: I think that makes sense, it does revert the decision in #4774, but we've recently decided that `main` docs reflect SNAPSHOT state anyways (for some reason those things connect for me, even though I'm not sure they really do)

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by trask** [comment]
@mateuszrzeszutek is your approval a vote to stay with versioned download link on `README.md` vs using `latest`?

**Comment by mateuszrzeszutek** [comment]
I think it makes slightly more sense to have versioned links -- READMEs on release branches and tags (that stem out from `main`) should have proper versions then.

**Comment by trask** [comment]
oh, I was not clear, @anuraaga's proposal is to keep versioned links on the tags, only using `latest` link on `main` (which would eliminate the need for bumping the version on `main`)

**Comment by mateuszrzeszutek** [comment]
Oh, so we'll replace the latest link with versioned link when the branch is cut? In that case, using `latest` seems to be a better choice 👍

**Comment by trask** [comment]
Closing, will send new PR to change to `latest` link on `main` branch, and update the release branch automation as needed for that change.

**Pull request closed without merging**